### PR TITLE
Refactor/normal gtp intercept

### PR DESCRIPTION
**Pull request by mrlukasbos**
@rolf can you check if the proper gotoposses are used? 

This removes the old pid controller entirely. Some small changes in  intercept where needed. i tested them in GRSim and it worked just fine.

### CODE REVIEW

File: roboteam_ai/src/skills/Skill.cpp [hunk 1]
```diff
@@ -15,11 +15,20 @@ void Skill::publishRobotCommand() {
     std::string ourSideParam;
     nh.getParam("our_side", ourSideParam);
 
-//    if(ourSideParam=="right"){
-//        cmd=rotateRobotCommand(cmd);
-//    }
-    ioManager.publishRobotCommand(command); // We default to our robots being on the left if parameter is not set
+    if(Constants::GRSIM() && ourSideParam=="right"){
```
Comment (rolfvdhulst) [inline roboteam_ai/src/skills/Skill.cpp]: This should be done in a seperate pr right? Why is this in here?

File: roboteam_ai/src/skills/Skill.cpp [hunk 2]
```diff
@@ -15,11 +15,20 @@ void Skill::publishRobotCommand() {
...
+      command=rotateRobotCommand(command);
+    }
 
+    if (command.id == -1) {
+        if (robot->id != -1) {
+            command.id = robot->id;
+            ioManager.publishRobotCommand(command); // We default to our robots being on the left if parameter is not set
+
```
Comment (rolfvdhulst) [inline roboteam_ai/src/skills/Skill.cpp]: unneccessary space

File: roboteam_ai/src/skills/Skill.cpp [hunk 3]
```diff
@@ -15,11 +15,20 @@ void Skill::publishRobotCommand() {
...
+        }
+    } else {
+        ioManager.publishRobotCommand(command); // We default to our robots being on the left if parameter is not set
+
```
Comment (rolfvdhulst) [inline roboteam_ai/src/skills/Skill.cpp]: unnecessary space

File: roboteam_ai/src/skills/InterceptBall.h [hunk 4]
```diff
@@ -35,14 +31,16 @@ class InterceptBall :public Skill {
...
+        control::NumTreePosControl numtreeGTP;
+        control::BasicPosControl basicGTP;
 
         Vector2 ballStartPos, ballStartVel, ballEndPos, interceptPos;
         Vector2 deltaPos;
         int tickCount, maxTicks;
-        control::PIDController pid,finePid;
         bool backwards;
 
+
```
Comment (rolfvdhulst) [inline roboteam_ai/src/skills/InterceptBall.h]: Why?

File: roboteam_ai/src/skills/Skill.cpp (hunk 1 above)
Comment (mrlukasbos) [reply roboteam_ai/src/skills/Skill.cpp]: this is now in development. needed it for 2 teams

**Pull request merged**